main/src/main/java/com/github/topi314/lavasrc/yandexmusic/YandexMusicSourceManager.java

```
@@ -148,7 +148,7 @@ private AudioItem getArtist(String id) throws IOException {
 
 	private AudioItem getPlaylist(String userString, String id) throws IOException {
 		var json = this.getJson(PUBLIC_API_BASE + "/users/" + userString + "/playlists/" + id);
-		if (json.isNull() || json.get("result").isNull() || json.get("result").get("tracks").values().isEmpty()) {
+		if (json == null || json.isNull() || !json.get("error").isNull() || json.get("result").isNull() || json.get("result").get("tracks").values().isEmpty()) {
```

here too

---

I didn't understand anything. I've only added json == null here and !json.get("error").isNull()

Since sometimes the response comes without json and json.isNull() returns an error

---

if you already check for an error response here `!json.get("error").isNull()` then you might as well throw an error when we get those.
silently returning no track on error would be kinda bad
before it would run into the code below and throw an error there since the other json props are missing, but this isnt the case anymore

---

Show an example of how to fix it

---

https://github.com/topi314/LavaSrc/blob/master/main/src/main/java/com/github/topi314/lavasrc/deezer/DeezerAudioTrack.java#L68